lib/gui/os/window-progress.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const electron = require('electron');
+const errors = require('../../shared/errors');
+
+/**
+ * @summary A reference to the current renderer Electron window
+ * @type {Object}
+ * @protected
+ *
+ * @description
+ * We expose this property to `this` for testability purposes.
+ */
+exports.currentWindow = electron.remote.getCurrentWindow();
+
+/**
+ * @summary Set operating system window progress
+ * @function
+ * @public
+ *
+ * @description
+ * Show progress inline in operating system task bar
+ *
+ * @param {Number} percentage - percentage
+ *
+ * @example
+ * windowProgress.set(85);
+ */
+exports.set = (percentage) => {
+  const PERCENTAGE_MINIMUM = 0;
+  const PERCENTAGE_MAXIMUM = 100;
+
+  if (percentage > PERCENTAGE_MAXIMUM || percentage < PERCENTAGE_MINIMUM) {
```

Comment:
There now seems to be several places throughout Etcher where we're checking that a percentage lies within the 0 -> 100 range, which leads to many re-definitions of `ZERO` and `HUNDRED`. Maybe it's time to add a shared `isValidPercentage` function?